src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -401,6 +405,79 @@ protected override void Dispose(bool disposing)
+
+        if (stateSampler != null)
+        {
+            sampler = stateSampler;
+        }
+
+        if (configuration.TryGetStringValue(TracesSamplerConfigKey, out var configValue))
+        {
+            if (sampler != null)
+            {
+                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent(
+                    $"Trace sampler configuration value '{configValue}' has been ignored because a value '{sampler.GetType().FullName}' was set previously.");
+                return sampler;
+            }
+
+            if (string.Equals(configValue, "always_on", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new AlwaysOnSampler();
+            }
+            else if (string.Equals(configValue, "always_off", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new AlwaysOffSampler();
+            }
+            else if (string.Equals(configValue, "traceidratio", StringComparison.OrdinalIgnoreCase))
+            {
+                var traceIdRatio = ReadTraceIdRatio(configuration);
+                sampler = new TraceIdRatioBasedSampler(traceIdRatio);
+            }
+            else if (string.Equals(configValue, "parentbased_always_on", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new ParentBasedSampler(new AlwaysOnSampler());
+            }
+            else if (string.Equals(configValue, "parentbased_always_off", StringComparison.OrdinalIgnoreCase))
+            {
+                sampler = new ParentBasedSampler(new AlwaysOffSampler());
+            }
+            else if (string.Equals(configValue, "parentbased_traceidratio", StringComparison.OrdinalIgnoreCase))
+            {
+                var traceIdRatio = ReadTraceIdRatio(configuration);
+                sampler = new ParentBasedSampler(new TraceIdRatioBasedSampler(traceIdRatio));
+            }
+            else
+            {
+                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent($"OTEL_TRACES_SAMPLER configuration was found but the value '{configValue}' is invalid and will be ignored.");
+            }
+
+            if (sampler != null)
+            {
+                OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent($"Trace sampler set to '{sampler.GetType().FullName}' from configuration.");
+            }
+        }
+
+        return sampler ?? new ParentBasedSampler(new AlwaysOnSampler());
+    }
+
+    private static double ReadTraceIdRatio(IConfiguration configuration)
+    {
+        if (configuration.TryGetStringValue(TracesSamplerArgConfigKey, out var configValue) &&
+                double.TryParse(configValue, out var traceIdRatio))
+        {
+            return traceIdRatio;
+        }
+        else
+        {
+            OpenTelemetrySdkEventSource.Log.TracerProviderSdkEvent($"Trace sampler argument configuration was found but the value '{configValue}' is invalid and will be ignored, default of value of '1.0' will be used.");
```

Comment:
same as https://github.com/open-telemetry/opentelemetry-dotnet/pull/5448/files#r1530819962
This could be another dedicated event. Ignoring user provided ratio is worth Warning level as the implications can be significant for users (billing/other-costs)